packages/docusaurus-mdx-loader/src/index.js

```diff
@@ -37,33 +45,41 @@ module.exports = async function docusaurusMdxLoader(fileString) {
 
   const hasFrontMatter = Object.keys(frontMatter).length > 0;
 
-  const options = {
-    ...reqOptions,
-    remarkPlugins: [
-      ...(reqOptions.beforeDefaultRemarkPlugins || []),
-      ...DEFAULT_OPTIONS.remarkPlugins,
-      [
-        transformImage,
-        {staticDir: reqOptions.staticDir, filePath: this.resourcePath},
+  if (!compilerCache.has(this.query)) {
+    const options = {
+      ...reqOptions,
+      remarkPlugins: [
+        ...(reqOptions.beforeDefaultRemarkPlugins || []),
+        ...DEFAULT_OPTIONS.remarkPlugins,
+        [
+          transformImage,
+          {staticDir: reqOptions.staticDir, filePath: this.resourcePath},
```

Comment:
Using `this.resourcePath` here makes me think this PR is not good enough.

Our remark plugins need to be able to access current file path to check if relative image/asset links are valid etc.

Reply:
Apparently we may be able to access it thanks to the `transform(node,vfile)` and `vfile.history[0]`: https://github.com/vfile/vfile#vfilehistory

Reply:
Yeah, I just took another look at https://github.com/mdx-js/mdx/pull/1468/files and they completely removed the `filePath: this.resourcePath` from the options object and only used it as parameter in the second argument of `compiler.process` - so it seems to be the accepted standard way to use that instead of accessing `filePath` from the options object.